tests/test_coordinator.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from mock import MagicMock, patch, mock_open
+
+import unittest
+import os
+import sys
+sys.path.append(os.path.dirname(os.path.realpath(__file__)))
+
+from package.scripts.presto_coordinator import Coordinator
+from test_worker import mock_file_descriptor_write_method
+
+class TestCoordinator(unittest.TestCase):
+
+    dummy_config_properties = {'pseudo.distributed.enabled': 'true',
+                               'query.queue-config-file': '',
+                               'http-server.http.port': '8081',
+                               'task.max-memory': '1GB'}
+
+    def setUp(self):
+        self.mock_env = MagicMock()
+
+    @patch('package.scripts.params.config_properties', new=dummy_config_properties)
+    @patch('package.scripts.presto_coordinator.smoketest_presto')
+    @patch('package.scripts.presto_coordinator.Coordinator.configure')
+    @patch('package.scripts.presto_coordinator.Execute')
+    def test_lifecycle_methods_shell_out_to_execute(
+            self, execute_mock, unused_configure_mock, unused_smoketest_presto):
+        presto_coordinator = Coordinator()
+
+        presto_coordinator.install(self.mock_env)
+        assert execute_mock.called
```

Comment:
I think you need to check some arguments here, because install can do the same as stop and test will pass.

I would suggest to check only some arguments (maybe just one) to not to make test too fragile. So I would check execute_mock was called with something related to install for presto_coordinator.install, related to stop for presto_coordinator.stop etc